tests/NuGet.Services.AzureSearch.Tests/DownloadTransferrerFacts.cs

```diff
@@ -28,6 +28,197 @@ public void ReturnsEmptyResult()
                 Assert.Empty(result);
             }
 
+            [Fact]
+            public void DoesNothingIfNoTransfers()
```

Comment:
Consider adding an integration test with a stateful, in-memory blob service to simulate from transitions like:

- No rename to A -> B rename
- A -> B rename to A -> C rename
- A -> B rename to B -> A rename
- A -> B rename to A -> B, C -> D rename

I think those state transitions are the most likely source for bugs.

Reply:
(this is in contrast to unit tests which may not cover the integration with the DAL)

Reply:
Added